File: 554-brick-wall/brick-wall.cpp
class Solution {
public:
    int leastBricks(vector<vector<int>>& wall) 
    {
        // {pos, count of Brick Edges}
        unordered_map<long long, int> map;
        int maxEdges = 0;
        for(int i = 0; i < wall.size(); i++)
        {
            long long pos = 0;
            // Exclude the last brick in each row
            for(int j = 0; j < wall[i].size() - 1; j++)
            {
                pos += wall[i][j];
                map[pos]++;
                maxEdges = max(maxEdges, map[pos]);
            }
        }
        return wall.size() - maxEdges;
    }
};

File: 554-brick-wall/brick-wall.java
class Solution {
    public int leastBricks(List<List<Integer>> wall) 
    {
        // {pos, count of Brick Edges}
        Map<Integer, Integer> map = new HashMap<>();
        int maxEdges = 0;
        for (List<Integer> row : wall) 
        {
            int pos = 0;
            // Exclude the last brick in each row
            for (int i = 0; i < row.size() - 1; i++) 
            {
                pos += row.get(i);
                map.put(pos, map.getOrDefault(pos, 0) + 1);
                maxEdges = Math.max(maxEdges, map.get(pos));
            }
        }
        return wall.size() - maxEdges;
    }
}